client/src/templates/Challenges/fill-in-the-blank/parse-blanks.ts

```diff
@@ -1,6 +1,46 @@
-type TextNode = { type: 'text'; value: string };
+type PlainTextNode = {
+  type: 'text';
+  value: string;
+};
+
+type HanziPinyinNode = {
+  type: 'hanzi-pinyin';
+  value: { hanzi: string; pinyin: string };
+};
+
 type BlankNode = { type: 'blank'; value: number };
-type ParagraphElement = TextNode | BlankNode;
+
+type ParagraphElement = PlainTextNode | HanziPinyinNode | BlankNode;
+
+const HANZI_PINYIN_PATTERN = /^(.+?)\s*\((.+?)\)$/;
```

Comment:
This is the exact regex we have in the challenge parser:

https://github.com/freeCodeCamp/freeCodeCamp/blob/3dc5cb6f43e232054ebf4ab2f94e5499daabfbff/tools/challenge-parser/parser/plugins/utils/i18n-stringify.js#L9

I realized just now that this regex is pretty naive, that it doesn't validate the characters and assumes anything follows the `text (text)` pattern is Chinese. 

I'll have a follow up PR to fix both this and the one in the challenge parser.